androguard/core/axml/__init__.py

```diff
@@ -988,7 +988,18 @@ def __init__(self, raw_buff: bytes)->bytes:
                         cur[-1].append(etree.Comment(comment))
 
                 logger.debug("START_TAG: {} (line={})".format(tag, self.axml.m_lineNumber))
-                elem = etree.Element(tag, nsmap=self.axml.nsmap)
+
+                try:
+                    elem = etree.Element(tag, nsmap=self.axml.nsmap)
+                except ValueError as e:
+                    logger.error(e)
+                    # nsmap= {'<!--': 'http://schemas.android.com/apk/res/android'} | pull/1056
+                    if 'Invalid namespace prefix' in str(e):
+                        corrected_nsmap = self.clean_and_replace_nsmap(self.axml.nsmap, str(e).split("'")[1])
```

Comment:
Parsing the output of an error message is very brittle as those could easily change.

Using `split("'")` to parse the output of `repr()` in such an error message even more so: e.g. it will definitely break if there's a `'` or any character that `repr()` will represent as a backslash-escape. Though it works for the given example.

Sadly, `lxml` doesn't seem to expose `libxml`'s `xmlValidateNameValue()` (which would only leave a check for `:`) or another function to easily check if a string is a valid prefix.

Reply:
Thank you @obfusk for taking a look at this. Indeed, I agree that error messages can easily change and this overly specific path could be irrelevant fast, though I believe it is a good approach to start with and adjust if we see things change. The logic behind it is very simple and would not require more than a few minutes to adjust it in the future.

Though its a good idea to write a test case for it, so it will be caught immediately if things changes.

Reply:
Also its a good idea when I will have enough time to further check which prefixes are allowed. I have already seen this method being used a few times.